### PR TITLE
Enable Parquet Writer test in AllTableWriterTest

### PR DESCRIPTION
**Pull request by majetideepak**
The PR that introduced https://github.com/facebookincubator/velox/pull/5055 missed some changes
after being imported. This resulted in the Parquet writer test AllTableWriterTest.directReadWrite being disabled.
A subsequent PR #5299 improved the AllTableWriterTest.directReadWrite test by adding bucketed and 
partitioned modes.
Parquet writer in these modes now fails with 
`NotImplemented: Writing DictionaryArray with null encoded in dictionary type not yet supported`
Only unpartitioned test mode works for the Parquet writer. The test has been modified for the same.

### CODE REVIEW

File: velox/exec/tests/TableWriteTest.cpp [hunk 1]
```diff
@@ -964,44 +964,36 @@ TEST_P(AllTableWriterTest, renameAndReorderColumns) {
 
 // Runs a pipeline with read + write.
 TEST_P(AllTableWriterTest, directReadWrite) {
-  for (const auto fileFormat :
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:967]: Can we make the file format a test parameter so it auto covered by all the table writer test cases? Thanks!

Comment (majetideepak) [reply velox/exec/tests/TableWriteTest.cpp:967]: @xiaoxmeng Not all the tests are working with Parquet due to https://github.com/facebookincubator/velox/issues/5560.
This PR is currently blocking https://github.com/facebookincubator/velox/pull/5469
Can we merge this to unblock the above PR?

Comment (majetideepak) [reply velox/exec/tests/TableWriteTest.cpp:967]: I will make this a test parameter.

**Comment by majetideepak** [comment]
@xiaoxmeng can you take a look now? Thanks.

File: velox/exec/tests/TableWriteTest.cpp [hunk 2]
```diff
@@ -660,103 +658,147 @@ class TableWriteTest : public HiveConnectorTestBase {
...
         false}
                              .value);
     testParams.push_back(TestParam{
+        FileFormat::DWRF,
         TestMode::kBucketed,
         CommitStrategy::kTaskCommit,
         HiveBucketProperty::Kind::kPrestoNative,
         false}
                              .value);
+    if (hasWriterFactory(FileFormat::PARQUET)) {
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:709]: Can we also cover bucketed table case?

File: velox/exec/tests/TableWriteTest.cpp [hunk 3]
```diff
@@ -660,103 +658,147 @@ class TableWriteTest : public HiveConnectorTestBase {
...
         false}
                              .value);
     testParams.push_back(TestParam{
+        FileFormat::DWRF,
         TestMode::kUnpartitioned,
         CommitStrategy::kTaskCommit,
         HiveBucketProperty::Kind::kHiveCompatible,
         false}
                              .value);
+    if (hasWriterFactory(FileFormat::PARQUET)) {
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:751]: ditto

**Review by xiaoxmeng — APPROVED** [review]
@majetideepak thanks for the update!

File: velox/exec/tests/TableWriteTest.cpp (hunk 2 above)
Comment (majetideepak) [reply velox/exec/tests/TableWriteTest.cpp:709]: There is an issue with partitioned and bucketed cases
See https://github.com/facebookincubator/velox/issues/5560
I will fix and add them.

**Pull request closed without merging**